### PR TITLE
Add resolveuid to Link content type `remoteUrl` field. Remove the variable interpolation support since it was broken anyways.

### PR DESCRIPTION
**Pull request by sneridagh**

### CODE REVIEW

**Review by ericof — APPROVED** [review]
LGTM

**Comment by sneridagh** [comment]
@jenkins-plone-org please run jobs

File: src/plone/restapi/tests/http-examples/link.resp [hunk 1]
```diff
@@ -1,66 +1,38 @@
-HTTP/1.1 200 OK
+HTTP/1.1 500 Internal Server Error
```
Comment (avoinea) [inline src/plone/restapi/tests/http-examples/link.resp:1]: Is this expected?

Comment (sneridagh) [reply src/plone/restapi/tests/http-examples/link.resp:1]: nope! :P

Comment (sneridagh) [reply src/plone/restapi/tests/http-examples/link.resp]: @avoinea Fixed

**Comment by sneridagh** [comment]
@jenkins-plone-org please run jobs

**Review by avoinea — APPROVED** [review]

**Comment by cekk** [comment]
i'm ok with these changes, but does this broke "classic Plone" compatibility?
I mean: if i create a Link in Volto, then in Plone everything works fine (the url resolver and the widget)? vice versa should works

**Comment by sneridagh** [comment]
@cekk I left the serializer in place, so if it has a "{portal_url}" it gets decoded. However, the compatibility was not complete either (navigation_root was not covered). I think it's better to rip it off from both sides.

**Comment by sneridagh** [comment]
@jenkins-plone-org please run jobs

File: src/plone/restapi/deserializer/blocks.py [hunk 2]
```diff
@@ -32,42 +29,6 @@ def iterate_children(value):
             queue.extend(child["children"] or [])
 
 
-def path2uid(context, link):
```
Comment (tisto) [inline src/plone/restapi/deserializer/blocks.py:35]: @sneridagh nitpicking: I usually prefer to keep util functions in the location where they are used. As long as a util function is only used in one location, moving it to its own file just makes it harder to understand what goes on.

**Review by tisto — APPROVED** [review]

File: src/plone/restapi/deserializer/blocks.py (hunk 2 above)
Comment (sneridagh) [reply src/plone/restapi/deserializer/blocks.py:35]: For the record, I moved them to its own module due to circular import issues.

**Review by mauritsvanrees — APPROVED** [review]
After careful consideration (see comments in issue #1197) this LGTM.

Perhaps change the news snippet, because variable interpolation support is not completely removed: restapi can still read such a value in both serialiser and deserialiser. It just saves it slightly differently.

**Comment by sneridagh** [comment]
@jenkins-plone-org please run jobs

File: src/plone/restapi/deserializer/blocks.py (hunk 2 above)
Comment (tiberiuichim) [reply src/plone/restapi/deserializer/blocks.py:35]: I'd argue that a Python library also has to consider the API it exposes as public. Would an import such as `from plone.restapi.blocks import path2uid` make sense? Not much. If the `path2uid` would be an internal util used only in the blocks.py module, then it should also be marked as internal, like `_path2uid`.

**Pull request merged**